Unpropagate gobjectIntrospection dependency from atk

There should really be no need to propagate `gobjectIntrospection`, it is only used for generating the gir files.

We should encounter only minor issues with packages like `gnome3.gnome-characters` or `gnome3.gnome-bluetooth`, which do not list the dependency explicitly.

Yet when I try to build the depending packages, `gnome3.gnome-screenshot` fails with the following, seemingly unrelated error:

```
ninja: Entering directory `build'
[1/7] gcc  -Isrc/gnome-screenshot@exe -Isrc -I../src -I. -I../ -I/nix/store/k95dqckfqbaq7lmnl5imdrnfh2nx67w9-xproto-7.0.31/include -I/nix/store/0r11wy9l82mac0nl77gzsrzs8hrqqphj-libX11-1.6.5-dev/include -I/nix/store/a3f86rv2pyn39vrkl7mxr5jdsvsnxa1h-kbproto-1.0.7/include -I/nix/store/13wsc4jwzz9lqzwhxw7rw7hcbzkxa8pn-libXext-1.3.3-dev/include -I/nix/store/vv8ykifz2g64085m7v2134n6jhq5d6qf-xextproto-7.3.0/include -I/nix/store/04j72yj4qiyxz8bjdh2hc1cny1mgzl79-glib-2.54.2-dev/include/glib-2.0 -I/nix/store/mqwwlkxw8g7i8sp7wpa88g9snvdbzh19-glib-2.54.2/lib/glib-2.0/include -I/nix/store/anviyjb4hmwp92awm0gfn0zm3wglvwd3-gtk+3-3.22.26-dev/include/gtk-3.0 -I/nix/store/m5gwd0qj963k8y53c7rhp21b9hhfqra5-cairo-1.14.10-dev/include/cairo -I/nix/store/lkb7h7jxrg8rp235r9yc2ig26590snva-freetype-2.7.1-dev/include/freetype2 -I/nix/store/lkb7h7jxrg8rp235r9yc2ig26590snva-freetype-2.7.1-dev/include -I/nix/store/zqrjh9m0q2ckpzl3q3jli5lh4pcp4nl8-pango-1.40.14-dev/include/pango-1.0 -I/nix/store/kxg0h8qmbgalzdj475bj5766kgw1g9vs-gdk-pixbuf-2.36.7-dev/include/gdk-pixbuf-2.0 -I/nix/store/qzkry5s1g6ijlxc8f0idnfac2i137scs-atk-2.26.1-dev/include/atk-1.0 -I/nix/store/n3i90lbpzc6kr5brdmi14xp7nxmvavfq-libcanberra-0.30/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O2 -g -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_DEFAULT_SOURCE -D_REENTRANT '-DLOCALEDIR="/usr/local/share/locale"' -DGLIB_DISABLE_DEPRECATION_WARNINGS -DGDK_DISABLE_DEPRECATION_WARNINGS -MMD -MQ 'src/gnome-screenshot@exe/screenshot-filename-builder.c.o' -MF 'src/gnome-screenshot@exe/screenshot-filename-builder.c.o.d' -o 'src/gnome-screenshot@exe/screenshot-filename-builder.c.o' -c ../src/screenshot-filename-builder.c
[2/7] /nix/store/y5brqifnrivf8f52a5j0acppi15iy2rf-python3-3.6.3/bin/python3.6m /nix/store/lwdnf1fm7x596gyak4x456gb59q25j7w-meson-0.43.0/bin/meson --internal msgfmthelper ../src/org.gnome.Screenshot.metainfo.xml.in src/org.gnome.Screenshot.metainfo.xml xml /tmp/nix-build-gnome-screenshot-3.26.0.drv-0/gnome-screenshot-3.26.0/src/../po
FAILED: src/org.gnome.Screenshot.metainfo.xml 
/nix/store/y5brqifnrivf8f52a5j0acppi15iy2rf-python3-3.6.3/bin/python3.6m /nix/store/lwdnf1fm7x596gyak4x456gb59q25j7w-meson-0.43.0/bin/meson --internal msgfmthelper ../src/org.gnome.Screenshot.metainfo.xml.in src/org.gnome.Screenshot.metainfo.xml xml /tmp/nix-build-gnome-screenshot-3.26.0.drv-0/gnome-screenshot-3.26.0/src/../po
msgfmt: cannot locate ITS rules for ../src/org.gnome.Screenshot.metainfo.xml.in
[3/7] gcc  -Isrc/gnome-screenshot@exe -Isrc -I../src -I. -I../ -I/nix/store/k95dqckfqbaq7lmnl5imdrnfh2nx67w9-xproto-7.0.31/include -I/nix/store/0r11wy9l82mac0nl77gzsrzs8hrqqphj-libX11-1.6.5-dev/include -I/nix/store/a3f86rv2pyn39vrkl7mxr5jdsvsnxa1h-kbproto-1.0.7/include -I/nix/store/13wsc4jwzz9lqzwhxw7rw7hcbzkxa8pn-libXext-1.3.3-dev/include -I/nix/store/vv8ykifz2g64085m7v2134n6jhq5d6qf-xextproto-7.3.0/include -I/nix/store/04j72yj4qiyxz8bjdh2hc1cny1mgzl79-glib-2.54.2-dev/include/glib-2.0 -I/nix/store/mqwwlkxw8g7i8sp7wpa88g9snvdbzh19-glib-2.54.2/lib/glib-2.0/include -I/nix/store/anviyjb4hmwp92awm0gfn0zm3wglvwd3-gtk+3-3.22.26-dev/include/gtk-3.0 -I/nix/store/m5gwd0qj963k8y53c7rhp21b9hhfqra5-cairo-1.14.10-dev/include/cairo -I/nix/store/lkb7h7jxrg8rp235r9yc2ig26590snva-freetype-2.7.1-dev/include/freetype2 -I/nix/store/lkb7h7jxrg8rp235r9yc2ig26590snva-freetype-2.7.1-dev/include -I/nix/store/zqrjh9m0q2ckpzl3q3jli5lh4pcp4nl8-pango-1.40.14-dev/include/pango-1.0 -I/nix/store/kxg0h8qmbgalzdj475bj5766kgw1g9vs-gdk-pixbuf-2.36.7-dev/include/gdk-pixbuf-2.0 -I/nix/store/qzkry5s1g6ijlxc8f0idnfac2i137scs-atk-2.26.1-dev/include/atk-1.0 -I/nix/store/n3i90lbpzc6kr5brdmi14xp7nxmvavfq-libcanberra-0.30/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O2 -g -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_DEFAULT_SOURCE -D_REENTRANT '-DLOCALEDIR="/usr/local/share/locale"' -DGLIB_DISABLE_DEPRECATION_WARNINGS -DGDK_DISABLE_DEPRECATION_WARNINGS -MMD -MQ 'src/gnome-screenshot@exe/screenshot-shadow.c.o' -MF 'src/gnome-screenshot@exe/screenshot-shadow.c.o.d' -o 'src/gnome-screenshot@exe/screenshot-shadow.c.o' -c ../src/screenshot-shadow.c
[4/7] gcc  -Isrc/gnome-screenshot@exe -Isrc -I../src -I. -I../ -I/nix/store/k95dqckfqbaq7lmnl5imdrnfh2nx67w9-xproto-7.0.31/include -I/nix/store/0r11wy9l82mac0nl77gzsrzs8hrqqphj-libX11-1.6.5-dev/include -I/nix/store/a3f86rv2pyn39vrkl7mxr5jdsvsnxa1h-kbproto-1.0.7/include -I/nix/store/13wsc4jwzz9lqzwhxw7rw7hcbzkxa8pn-libXext-1.3.3-dev/include -I/nix/store/vv8ykifz2g64085m7v2134n6jhq5d6qf-xextproto-7.3.0/include -I/nix/store/04j72yj4qiyxz8bjdh2hc1cny1mgzl79-glib-2.54.2-dev/include/glib-2.0 -I/nix/store/mqwwlkxw8g7i8sp7wpa88g9snvdbzh19-glib-2.54.2/lib/glib-2.0/include -I/nix/store/anviyjb4hmwp92awm0gfn0zm3wglvwd3-gtk+3-3.22.26-dev/include/gtk-3.0 -I/nix/store/m5gwd0qj963k8y53c7rhp21b9hhfqra5-cairo-1.14.10-dev/include/cairo -I/nix/store/lkb7h7jxrg8rp235r9yc2ig26590snva-freetype-2.7.1-dev/include/freetype2 -I/nix/store/lkb7h7jxrg8rp235r9yc2ig26590snva-freetype-2.7.1-dev/include -I/nix/store/zqrjh9m0q2ckpzl3q3jli5lh4pcp4nl8-pango-1.40.14-dev/include/pango-1.0 -I/nix/store/kxg0h8qmbgalzdj475bj5766kgw1g9vs-gdk-pixbuf-2.36.7-dev/include/gdk-pixbuf-2.0 -I/nix/store/qzkry5s1g6ijlxc8f0idnfac2i137scs-atk-2.26.1-dev/include/atk-1.0 -I/nix/store/n3i90lbpzc6kr5brdmi14xp7nxmvavfq-libcanberra-0.30/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O2 -g -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_DEFAULT_SOURCE -D_REENTRANT '-DLOCALEDIR="/usr/local/share/locale"' -DGLIB_DISABLE_DEPRECATION_WARNINGS -DGDK_DISABLE_DEPRECATION_WARNINGS -MMD -MQ 'src/gnome-screenshot@exe/screenshot-dialog.c.o' -MF 'src/gnome-screenshot@exe/screenshot-dialog.c.o.d' -o 'src/gnome-screenshot@exe/screenshot-dialog.c.o' -c ../src/screenshot-dialog.c
[5/7] gcc  -Isrc/gnome-screenshot@exe -Isrc -I../src -I. -I../ -I/nix/store/k95dqckfqbaq7lmnl5imdrnfh2nx67w9-xproto-7.0.31/include -I/nix/store/0r11wy9l82mac0nl77gzsrzs8hrqqphj-libX11-1.6.5-dev/include -I/nix/store/a3f86rv2pyn39vrkl7mxr5jdsvsnxa1h-kbproto-1.0.7/include -I/nix/store/13wsc4jwzz9lqzwhxw7rw7hcbzkxa8pn-libXext-1.3.3-dev/include -I/nix/store/vv8ykifz2g64085m7v2134n6jhq5d6qf-xextproto-7.3.0/include -I/nix/store/04j72yj4qiyxz8bjdh2hc1cny1mgzl79-glib-2.54.2-dev/include/glib-2.0 -I/nix/store/mqwwlkxw8g7i8sp7wpa88g9snvdbzh19-glib-2.54.2/lib/glib-2.0/include -I/nix/store/anviyjb4hmwp92awm0gfn0zm3wglvwd3-gtk+3-3.22.26-dev/include/gtk-3.0 -I/nix/store/m5gwd0qj963k8y53c7rhp21b9hhfqra5-cairo-1.14.10-dev/include/cairo -I/nix/store/lkb7h7jxrg8rp235r9yc2ig26590snva-freetype-2.7.1-dev/include/freetype2 -I/nix/store/lkb7h7jxrg8rp235r9yc2ig26590snva-freetype-2.7.1-dev/include -I/nix/store/zqrjh9m0q2ckpzl3q3jli5lh4pcp4nl8-pango-1.40.14-dev/include/pango-1.0 -I/nix/store/kxg0h8qmbgalzdj475bj5766kgw1g9vs-gdk-pixbuf-2.36.7-dev/include/gdk-pixbuf-2.0 -I/nix/store/qzkry5s1g6ijlxc8f0idnfac2i137scs-atk-2.26.1-dev/include/atk-1.0 -I/nix/store/n3i90lbpzc6kr5brdmi14xp7nxmvavfq-libcanberra-0.30/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O2 -g -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_DEFAULT_SOURCE -D_REENTRANT '-DLOCALEDIR="/usr/local/share/locale"' -DGLIB_DISABLE_DEPRECATION_WARNINGS -DGDK_DISABLE_DEPRECATION_WARNINGS -MMD -MQ 'src/gnome-screenshot@exe/screenshot-utils.c.o' -MF 'src/gnome-screenshot@exe/screenshot-utils.c.o.d' -o 'src/gnome-screenshot@exe/screenshot-utils.c.o' -c ../src/screenshot-utils.c
[6/7] gcc  -Isrc/gnome-screenshot@exe -Isrc -I../src -I. -I../ -I/nix/store/k95dqckfqbaq7lmnl5imdrnfh2nx67w9-xproto-7.0.31/include -I/nix/store/0r11wy9l82mac0nl77gzsrzs8hrqqphj-libX11-1.6.5-dev/include -I/nix/store/a3f86rv2pyn39vrkl7mxr5jdsvsnxa1h-kbproto-1.0.7/include -I/nix/store/13wsc4jwzz9lqzwhxw7rw7hcbzkxa8pn-libXext-1.3.3-dev/include -I/nix/store/vv8ykifz2g64085m7v2134n6jhq5d6qf-xextproto-7.3.0/include -I/nix/store/04j72yj4qiyxz8bjdh2hc1cny1mgzl79-glib-2.54.2-dev/include/glib-2.0 -I/nix/store/mqwwlkxw8g7i8sp7wpa88g9snvdbzh19-glib-2.54.2/lib/glib-2.0/include -I/nix/store/anviyjb4hmwp92awm0gfn0zm3wglvwd3-gtk+3-3.22.26-dev/include/gtk-3.0 -I/nix/store/m5gwd0qj963k8y53c7rhp21b9hhfqra5-cairo-1.14.10-dev/include/cairo -I/nix/store/lkb7h7jxrg8rp235r9yc2ig26590snva-freetype-2.7.1-dev/include/freetype2 -I/nix/store/lkb7h7jxrg8rp235r9yc2ig26590snva-freetype-2.7.1-dev/include -I/nix/store/zqrjh9m0q2ckpzl3q3jli5lh4pcp4nl8-pango-1.40.14-dev/include/pango-1.0 -I/nix/store/kxg0h8qmbgalzdj475bj5766kgw1g9vs-gdk-pixbuf-2.36.7-dev/include/gdk-pixbuf-2.0 -I/nix/store/qzkry5s1g6ijlxc8f0idnfac2i137scs-atk-2.26.1-dev/include/atk-1.0 -I/nix/store/n3i90lbpzc6kr5brdmi14xp7nxmvavfq-libcanberra-0.30/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O2 -g -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_DEFAULT_SOURCE -D_REENTRANT '-DLOCALEDIR="/usr/local/share/locale"' -DGLIB_DISABLE_DEPRECATION_WARNINGS -DGDK_DISABLE_DEPRECATION_WARNINGS -MMD -MQ 'src/gnome-screenshot@exe/screenshot-interactive-dialog.c.o' -MF 'src/gnome-screenshot@exe/screenshot-interactive-dialog.c.o.d' -o 'src/gnome-screenshot@exe/screenshot-interactive-dialog.c.o' -c ../src/screenshot-interactive-dialog.c
ninja: build stopped: subcommand failed.
```

We need to find out why this error happens before removing the dependency.

--- a/pkgs/development/libraries/gettext/default.nix
+++ b/pkgs/development/libraries/gettext/default.nix
@@ -48,6 +48,8 @@ stdenv.mkDerivation rec {
   # HACK, see #10874 (and 14664)
   buildInputs = stdenv.lib.optional (!stdenv.isLinux && !hostPlatform.isCygwin) libiconv;
 
+  setupHook = ./gettext-setup-hook.sh;
+
   enableParallelBuilding = true;
 
   meta = {

--- a/pkgs/development/libraries/gettext/gettext-setup-hook.sh
+++ b/pkgs/development/libraries/gettext/gettext-setup-hook.sh
@@ -1,0 +1,7 @@
+gettextDataDirsHook() {
+    if [ -d "$1/share/gettext" ]; then
+        addToSearchPath GETTEXTDATADIRS "$1/share/gettext"
+    fi
+}
+
+envHooks+=(gettextDataDirsHook)

